src/Microsoft.ML.Data/Model/Onnx/SaveOnnxCommand.cs

```diff
@@ -228,10 +266,68 @@ private void Run(IChannel ch)
 
             if (!string.IsNullOrWhiteSpace(Args.OutputModelFile))
             {
+                Contracts.Assert(loader != null);
+
                 ch.Trace("Saving the data pipe");
                 // Should probably include "end"?
                 SaveLoader(loader, Args.OutputModelFile);
             }
         }
+
+        public sealed class Output
+        {
+            //REVIEW: Would be nice to include ONNX protobuf model here but code generator needs an upgrade.
```

Comment:
>//REVIEW: Would be nice to include ONNX protobuf model here but code generator needs an upgrade. [](start = 12, length = 96)

We ought not to include the model *itself* (if by that you mean the compiled protobuf object) since it has a protobuf dependency. And while the code *can* include a dependency on that, we don't want the API layer having a dependency on that, since there is some hope we might be able to factor out the protobuf into a separate assembly, somehow.

Also: despite the model file not being totally super-duper up-to-date, you ought to sketch out how you view this thing as working. Otherwise we have no way to review your design, since you've left that as a todo, but there is nothing preventing you from producing actual output right now (just as the ONNX command produces output right now).

Reply:
Just want to clarify, are you saying we should include JSON in the output? because that is written to the file specified in the input and we can use that to validate?

---
In reply to: [191824117](https://github.com/dotnet/machinelearning/pull/248#discussion_r191824117) [](ancestors = 191824117)

Reply:
I'm saying only we ought not to include the protobuf object, then explaining why. I'm not saying anything else. What you describe may or may not be a good idea, but not addressing that here.

---
In reply to: [191901399](https://github.com/dotnet/machinelearning/pull/248#discussion_r191901399) [](ancestors = 191901399,191824117)

Reply:
Understood, thanks!

---
In reply to: [191916172](https://github.com/dotnet/machinelearning/pull/248#discussion_r191916172) [](ancestors = 191916172,191901399,191824117)